### PR TITLE
snakeviz: test with cProfile output

### PR DESCRIPTION
**Pull request by rtau**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
The more realistic test as mentioned in https://github.com/Homebrew/homebrew-core/pull/155280#issuecomment-1825940694

### CODE REVIEW

**Comment by chenrui333** [comment]
`    TypeError: no implicit conversion from nil to integer`

File: Formula/s/snakeviz.rb [hunk 1]
```diff
@@ -30,15 +31,20 @@ def install
 
   test do
     system "#{bin}/snakeviz", "--version"
+    system "python3", "--version"
+    system "python", "--version"
+    system "python3", "-m", "cProfile", "-o", "output.prof", "-m", "cProfile"
```
Comment (Bo98) [inline Formula/s/snakeviz.rb:36]: There isn't a `python3` in `python@3.12` yet but `python3.12` will work here.

File: Formula/s/snakeviz.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
+require "cgi"
```
Comment (Bo98) [inline Formula/s/snakeviz.rb:1]: Can we lazy load this by moving this to the start of the `test` block instead?

Comment (rtau) [reply Formula/s/snakeviz.rb:1]: Yes, that should be possible.

File: Formula/s/snakeviz.rb (hunk 1 above)
Comment (rtau) [reply Formula/s/snakeviz.rb]: Thanks. Test on Linux pass now.

**Comment by rtau** [comment]
@chenrui333 test failures had been fixed.

**Review by chenrui333 — APPROVED** [review]
lgtm, thanks!

**Pull request merged**